# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/PHttpChannel.ipdl
# Commit: c7862a00e6e7
# Full Hash: c7862a00e6e74fec00119ab746639ead6954b702
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-19 03:53:26
# Regressor Bug: 1839918
# File Overlap Count: 1
# Description:
#   Bug 1839918 - Use PHttpChannel for SetCookies calls from child channels. r=bvandersloot,necko-reviewers,cookie-reviewers,valentin
#   
#   Depends on D186045
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D186340
# ==============================================================================

diff -r 8d7f44b9b6e2 -r c7862a00e6e7 netwerk/protocol/http/PHttpChannel.ipdl
--- a/netwerk/protocol/http/PHttpChannel.ipdl	Mon Sep 18 11:55:09 2023 +0000
+++ b/netwerk/protocol/http/PHttpChannel.ipdl	Mon Sep 18 11:55:10 2023 +0000
@@ -62,6 +62,15 @@
                                       PrincipalInfo requestingPrincipal,
                                       OriginAttributes originAttributes);
 
+  // Send cookies to the parent for a given channel. Compared to PCookieService
+  // SetCookies this method allows the parent to determine which BrowsingContext
+  // the request was sent for.
+  async SetCookies(nsCString baseDomain,
+                                    OriginAttributes attrs,
+                                    nullable nsIURI host,
+                                    bool fromHttp,
+                                    CookieStruct[] cookies);
+
   // After receiving this message, the parent calls SendDeleteSelf, and makes
   // sure not to send any more messages after that.
   async DeletingChannel();
